src/basic/fd-util.c

```diff
@@ -864,3 +865,52 @@ int fd_get_diskseq(int fd, uint64_t *ret) {
 
         return 0;
 }
+
+int dir_fd_is_root(int dir_fd) {
+        STRUCT_NEW_STATX_DEFINE(st);
+        STRUCT_NEW_STATX_DEFINE(pst);
+        int r;
+
+        assert(dir_fd >= 0);
+
+        r = statx_fallback(dir_fd, "", AT_EMPTY_PATH, STATX_TYPE|STATX_INO|STATX_MNT_ID, &st.sx);
+        if (r < 0)
```

Comment:
why use AT_EMPTY_PATH here instead of "."?

if you use `"."` you could verify as side-effect that this is a dir. And if you get ENOTDIR then you just turn that into `return false`. Saves you a syscall